src/utilities/attachHasInstanceSymbol.js

```diff
@@ -0,0 +1,25 @@
+/* @flow */
+/**
+ *  Copyright (c) 2015, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+// eslint-disable-next-line flowtype/no-weak-types
+export default function attachHasInstanceSymbol(ctor: Function): void {
+  if (typeof Symbol === 'undefined' || !Symbol.for || !Symbol.hasInstance) {
+    return;
+  }
+  const tag = `@@typeof/${ctor.name}`;
+
+  Object.defineProperty(ctor, Symbol.hasInstance, {
+    value: function $hasInstance(instance) {
+      return instance && instance[Symbol.for(tag)] === true;
+    },
+  });
+
+  ctor.prototype[Symbol.for(tag)] = true;
```

Comment:
Why `Symbol.for()` and not `Symbol()`?

Reply:
one fetches a global symbol by name vs creating a new symbol:

```js
Symbol('foo') == Symbol('foo') // false
Symbol.for('foo') == Symbol.for('foo') // true
```

Reply:
I understand, but why is that desirable here? Imo the symbol should be private or if it's public, it should be exported, no retrievable for everyone through a magic string. You just need to save the symbol in a variable and reuse it.

Reply:
Its not using a symbol for privacy, it's to avoid mudding up people's namespace on the class for things they can use as property names. With the symbol you don't have to worry (nearly as much) about accidentally overwriting the tag in an instance when you just wanted to define an unrelated property. Incidentally this approach is how React tags elements as "ReactElements"